### DISCUSSION TITLE
Enable a generic postprocessing

### DISCUSSION THREAD

**Discussion by aborruso** [githubocto/flat-action#14] [Ideas & Feature Requests]
I working a lot using git scraping, actions and bash postprocessing.

In [this very basic example](https://github.com/aborruso/rss/blob/main/.github/workflows/ordinanzeCovidRegioneSiciliana.yml)  I use this [bash file](https://github.com/aborruso/rss/blob/main/script/ordinanzeCovidRegioneSiciliana/ordinanzeCovidRegioneSiciliana.sh) .

It would be very convenient to enable the use of a generic postprocessing file: `postprocess: ./helloWorld.sh`

Thank you this is a great project

**Comment by aborruso** [comment]
here a bash postprocessing example repo https://github.com/aborruso/flat_data_bash_example

**Reply by mskyttner** [reply]
Fantastic!

Do you think it would be possible to run docker commands from a bash-file to do postprocessing?

**Reply by aborruso** [reply]
> Do you think it would be possible to run docker commands from a bash-file to do postprocessing?

I think yes!

**Reply by mskyttner** [reply]
Thanks for your example! 

I tried to fork your repo and tweak it to do the same operation your bash script does but with an Rscript (bash and python are pre-installed on ubuntu latest but R is not).

It seems to work when Deno is installed inside a container which already has the dependency present (R, miller, or something else), which means Deno gets access to R which is present already in that container and can run that process: https://github.com/mskyttner/flat_data_bash_example/blob/main/.github/workflows/flat.yml#L13). 

So it is not quite calling a docker command from Deno but rather putting Deno in a docker container which has the required tool/dependency present. Then Deno can shell out to run the postprocessing script: https://github.com/mskyttner/flat_data_bash_example/blob/main/postprocessing.R.

**Reply by aborruso** [reply]
Very useful!

**Reply by mskyttner** [reply]
Still not sure how to shell out to run a docker or podman command. It would be awesome to have an example of a bash script "shelling out" to a docker or podman command that would do some processing of data. But it would be running a container from within a container. Perhaps possible if using a container like mgoltzsche/podman if the Flat action can run on it....
